test/runner/reporter.js

```diff
@@ -57,30 +70,35 @@ export function reportTest( test, reportId, { browser, headless } ) {
 						diff = [ { removed: true, value: `${ value }` } ];
 					}
 				} else if (
-					typeof error.expected === "boolean" &&
-					typeof error.actual === "boolean"
+					typeof error.expected === "string" &&
+					typeof error.actual === "string"
 				) {
 
-					// Show the actual boolean in red
-					diff = [ { removed: true, value: `${ error.actual }` } ];
-				} else {
+					// Diff the characters of strings
+					diff = Diff.diffChars( error.expected, error.actual );
+				} else if ( error.expected != null && error.actual != null ) {
 
-					// Diff everything else as characters
-					diff = Diff.diffChars( `${ error.expected }`, `${ error.actual }` );
+					// Diff everything else as words
+					diff = Diff.diffWords(
+						quoteWrap( error.expected ),
+						quoteWrap( error.actual )
+					);
```

Comment:
I think the necessary insight here is that we want to diff type-mismatched values _after serialization_, which `quoteWrap` a) fails to convey, and b) currently does so inconsistently (e.g., it escapes `"` in strings by `\`-prefixing, but does not similarly escape `\`).
```suggestion
					// Diff everything else as words (post-serialization)
					diff = Diff.diffWords(
						serializeForDiff( error.expected ),
						serializeForDiff( error.actual )
					);
```
with
```js
function serializeForDiff( value ) {
	// Use naive serialization for everything except types with confusable values
	if ( typeof value === "string" ) {
		return JSON.stringify(value);
	} else if ( typeof value === "bigint" ) {
		return `${ value }n`;
	} else if ( typeof value === "symbol" ) {
		const ctor = Symbol.keyFor(value) !== undefined ? "Symbol.for" : "Symbol";
		return `${ctor}(${JSON.stringify(value.description)})`;
	}
	return `${ value }`;
}
```

Reply:
I like that solution.

Reply:
I like that, too! Using `JSON.stringify` on strings is a great idea. ❤️

Reply:
I'll try it and post some examples.

Reply:
Something I realized while testing is that Symbols do not get sent over the wire (they get removed when passed through `JSON.stringify`. So, I moved that part to listeners.js.

<img width="598" alt="image" src="https://github.com/jquery/jquery/assets/192451/1d24a835-922f-4c0d-8f8a-742e6985fe4f">
<img width="130" alt="image" src="https://github.com/jquery/jquery/assets/192451/efb98a8c-f08e-40c9-952c-26c1589e3037">

Reply:
Also, I switched to just show the diff all the time when at least one is non-null. I found I preferred to see it as confirmation.